universal-application-tool-0.0.1/test/controllers/applicant/ApplicantProgramsControllerTest.java

```diff
@@ -49,13 +62,40 @@ public void index_withPrograms_returnsAllPrograms() {
   public void index_withProgram_includesApplyButtonWithRedirect() {
     Program program = resourceCreator().insertProgram("program");
 
-    Result result = controller.index(1L).toCompletableFuture().join();
+    Result result = controller.index(fakeRequest().build(), 1L).toCompletableFuture().join();
 
     assertThat(result.status()).isEqualTo(OK);
     assertThat(contentAsString(result))
         .contains(routes.ApplicantProgramsController.edit(1L, program.id).url());
   }
 
+  @Test
+  public void index_usesMessagesForUserPreferredLocale() {
+    // Stub the MessagesApi with fake translations.
+    Langs langs = new Langs(new DefaultLangs());
+    Map<String, String> translations = Collections.singletonMap("programs", "Different");
+    Map<String, Map<String, String>> messageMap =
+        ImmutableMap.of(Lang.defaultLang().code(), translations);
+    MessagesApi messagesApi = stubMessagesApi(messageMap, langs);
```

Comment:
Issue (blocking): this is a decent amount of setup and probably more than we want to do for each controller test once we've localized the whole app. How might we reduce the level of effort for testing controllers that render localized views?

Reply:
I added a `LocalizationHelper` to the controller test directory for now, but we may want to update `WithPostgresContainer` or some other test class with the override.

Alternatively, it occurs to me we can simply set a different language on the request (using a simple `stubMessageApi()`) and then use the actual translations - which would you prefer?

Reply:
The latter sounds really good -- then we'd be testing more of the real codepath.

Reply:
Done! Much simpler